diff --git a/core/base.go b/core/base.go
index 4551e3f..d1ca96c 100644
--- a/core/base.go
+++ b/core/base.go
@@ -17,7 +17,6 @@ import (
 	"github.com/fatih/color"
 	"github.com/pocketbase/dbx"
 	"github.com/pocketbase/pocketbase/tools/cron"
-	"github.com/pocketbase/pocketbase/tools/filesystem"
 	"github.com/pocketbase/pocketbase/tools/hook"
 	"github.com/pocketbase/pocketbase/tools/logger"
 	"github.com/pocketbase/pocketbase/tools/mailer"
@@ -399,7 +398,7 @@ func (app *BaseApp) Bootstrap() error {
 		}

 		// ensure that data dir exist
-		if err := os.MkdirAll(app.DataDir(), os.ModePerm); err != nil {
+		if err := app.prepareDataDir(); err != nil {
 			return err
 		}

@@ -427,9 +426,6 @@ func (app *BaseApp) Bootstrap() error {
 			return err
 		}

-		// try to cleanup the pb_data temp directory (if any)
-		_ = os.RemoveAll(filepath.Join(app.DataDir(), LocalTempDirName))
-
 		return nil
 	})

@@ -706,49 +702,6 @@ func (app *BaseApp) NewMailClient() mailer.Mailer {
 	return client
 }

-// NewFilesystem creates a new local or S3 filesystem instance
-// for managing regular app files (ex. record uploads)
-// based on the current app settings.
-//
-// NB! Make sure to call Close() on the returned result
-// after you are done working with it.
-func (app *BaseApp) NewFilesystem() (*filesystem.System, error) {
-	if app.settings != nil && app.settings.S3.Enabled {
-		return filesystem.NewS3(
-			app.settings.S3.Bucket,
-			app.settings.S3.Region,
-			app.settings.S3.Endpoint,
-			app.settings.S3.AccessKey,
-			app.settings.S3.Secret,
-			app.settings.S3.ForcePathStyle,
-		)
-	}
-
-	// fallback to local filesystem
-	return filesystem.NewLocal(filepath.Join(app.DataDir(), LocalStorageDirName))
-}
-
-// NewBackupsFilesystem creates a new local or S3 filesystem instance
-// for managing app backups based on the current app settings.
-//
-// NB! Make sure to call Close() on the returned result
-// after you are done working with it.
-func (app *BaseApp) NewBackupsFilesystem() (*filesystem.System, error) {
-	if app.settings != nil && app.settings.Backups.S3.Enabled {
-		return filesystem.NewS3(
-			app.settings.Backups.S3.Bucket,
-			app.settings.Backups.S3.Region,
-			app.settings.Backups.S3.Endpoint,
-			app.settings.Backups.S3.AccessKey,
-			app.settings.Backups.S3.Secret,
-			app.settings.Backups.S3.ForcePathStyle,
-		)
-	}
-
-	// fallback to local filesystem
-	return filesystem.NewLocal(filepath.Join(app.DataDir(), LocalBackupsDirName))
-}
-
 // Restart restarts (aka. replaces) the current running application process.
 //
 // NB! It relies on execve which is supported only on UNIX based systems.
diff --git a/core/notify_watcher.go b/core/notify_watcher.go
index 1f548cd..90b700b 100644
--- a/core/notify_watcher.go
+++ b/core/notify_watcher.go
@@ -1,3 +1,5 @@
+//go:build !js
+
 package core

 import (
