diff --git a/migrations/1640988000_init.go b/migrations/1640988000_init.go
index 541f045..e89e276 100644
--- a/migrations/1640988000_init.go
+++ b/migrations/1640988000_init.go
@@ -35,7 +35,7 @@ func init() {
 		// -----------------------------------------------------------
 
 		_, execerr := txApp.DB().NewQuery(`
-			CREATE TABLE {{_collections}} (
+			CREATE TABLE IF NOT EXISTS {{_collections}} (
 				[[id]]         TEXT PRIMARY KEY DEFAULT ('r'||lower(hex(randomblob(7)))) NOT NULL,
 				[[system]]     BOOLEAN DEFAULT FALSE NOT NULL,
 				[[type]]       TEXT DEFAULT "base" NOT NULL,
@@ -106,7 +106,7 @@ func init() {
 
 func createParamsTable(txApp core.App) error {
 	_, execErr := txApp.DB().NewQuery(`
-		CREATE TABLE {{_params}} (
+		CREATE TABLE IF NOT EXISTS {{_params}} (
 			[[id]]      TEXT PRIMARY KEY DEFAULT ('r'||lower(hex(randomblob(7)))) NOT NULL,
 			[[value]]   JSON DEFAULT NULL,
 			[[created]] TEXT DEFAULT (strftime('%Y-%m-%d %H:%M:%fZ')) NOT NULL,
