--- a/wasm_exec.js
+++ b/wasm_exec.js
@@ -476,19 +476,24 @@
 			};
 		}
 
-		async run(instance) {
+		async run(instance, context) {
 			if (!(instance instanceof WebAssembly.Instance)) {
 				throw new Error("Go.run: WebAssembly.Instance expected");
 			}
 			this._inst = instance;
 			this.mem = new DataView(this._inst.exports.mem.buffer);
+			const globalProxy = new Proxy(globalThis, {
+				get(target, prop) {
+					return prop === "context" ? context : Reflect.get(target, prop, target);
+				},
+			});
 			this._values = [ // JS values that Go currently has references to, indexed by reference id
 				NaN,
 				0,
 				null,
 				true,
 				false,
-				globalThis,
+				globalProxy,
 				this,
 			];
 			this._goRefCounts = new Array(this._values.length).fill(Infinity); // number of references that Go has to a JS value, indexed by reference id
@@ -497,7 +502,7 @@
 				[null, 2],
 				[true, 3],
 				[false, 4],
-				[globalThis, 5],
+				[globalProxy, 5],
 				[this, 6],
 			]);
 			this._idPool = [];   // unused ids that have been garbage collected
